src/rules/sort-collections.ts

```diff
@@ -29,14 +29,38 @@ export const rule = createRule<Options>({
 					toSort.includes(key.value)
 				) {
 					const currentOrder = collection.properties;
-					const desiredOrder = currentOrder
-						.slice()
-						.sort((a, b) =>
-							(a.key as AST.JSONStringLiteral).value >
-							(b.key as AST.JSONStringLiteral).value
-								? 1
-								: -1,
-						);
+					const desiredOrder = currentOrder.slice().sort((a, b) => {
+						let aKey = (a.key as AST.JSONStringLiteral).value;
+						let bKey = (b.key as AST.JSONStringLiteral).value;
+
+						if (key.value !== "scripts") {
```

Comment:
[Refactor] We know whether the `key.value` is `"scripts"` before this function is declared. So instead of checking _inside_ the function:

```ts
.sort((a, b) => {
  if (key.value !== "scripts") {
    // non-scripts logic
  } else {
    // scripts-specific logic
  }
})
```

...I suspect it might be cleaner to pass `.sort()` a different function based on `key.value`:

```ts
.sort(
  key.value === "scripts"
    ? (a, b) => { /* scripts-specific logic */ }
    : (a, b) => { { scripts-specific logic */ }
)
```

I'm not sure. Maybe it'd be super ugly and extra code unnecessarily. What do you think about trying this out and seeing how it looks?

Reply:
The problem is that `aKey` and `bKey` are used in both. So it'll add duplicate code as `aKey` and `bKey` will be defined twice.